src/evm_context.rs

```diff
@@ -109,191 +105,160 @@ context_variables! {
-        ssa,
-        &context().revert_flag.identifier,
-        "#x0000000000000000000000000000000000000000000000000000000000000001",
-    )
+pub fn set_reverted(ssa: &mut SSATracker) -> SMTStatement {
+    assign_variable_if_executing_regularly(ssa, &context().revert_flag.identifier, 1.into())
 }
 
-pub fn set_stopped(ssa: &mut SSATracker) -> String {
-    assign_variable_if_executing_regularly(
-        ssa,
-        &context().stop_flag.identifier,
-        "#x0000000000000000000000000000000000000000000000000000000000000001",
-    )
+pub fn set_stopped(ssa: &mut SSATracker) -> SMTStatement {
+    assign_variable_if_executing_regularly(ssa, &context().stop_flag.identifier, 1.into())
 }
 
-pub fn address(ssa: &mut SSATracker) -> String {
+pub fn address(ssa: &mut SSATracker) -> SMTVariable {
     context().address.smt_var(ssa)
 }
-pub fn origin(ssa: &mut SSATracker) -> String {
+pub fn origin(ssa: &mut SSATracker) -> SMTVariable {
     context().origin.smt_var(ssa)
 }
-pub fn caller(ssa: &mut SSATracker) -> String {
+pub fn caller(ssa: &mut SSATracker) -> SMTVariable {
     context().caller.smt_var(ssa)
 }
-pub fn callvalue(ssa: &mut SSATracker) -> String {
+pub fn callvalue(ssa: &mut SSATracker) -> SMTVariable {
     context().callvalue.smt_var(ssa)
 }
-pub fn calldatasize(ssa: &mut SSATracker) -> String {
+pub fn calldatasize(ssa: &mut SSATracker) -> SMTVariable {
     context().calldatasize.smt_var(ssa)
 }
-pub fn codesize(ssa: &mut SSATracker) -> String {
+pub fn codesize(ssa: &mut SSATracker) -> SMTVariable {
     context().codesize.smt_var(ssa)
 }
-pub fn gasprice(ssa: &mut SSATracker) -> String {
+pub fn gasprice(ssa: &mut SSATracker) -> SMTVariable {
     context().gasprice.smt_var(ssa)
 }
-pub fn coinbase(ssa: &mut SSATracker) -> String {
+pub fn coinbase(ssa: &mut SSATracker) -> SMTVariable {
     context().coinbase.smt_var(ssa)
 }
-pub fn timestamp(ssa: &mut SSATracker) -> String {
+pub fn timestamp(ssa: &mut SSATracker) -> SMTVariable {
     context().timestamp.smt_var(ssa)
 }
-pub fn number(ssa: &mut SSATracker) -> String {
+pub fn number(ssa: &mut SSATracker) -> SMTVariable {
     context().number.smt_var(ssa)
 }
-pub fn difficulty(ssa: &mut SSATracker) -> String {
+pub fn difficulty(ssa: &mut SSATracker) -> SMTVariable {
     context().difficulty.smt_var(ssa)
 }
-pub fn gaslimit(ssa: &mut SSATracker) -> String {
+pub fn gaslimit(ssa: &mut SSATracker) -> SMTVariable {
     context().gaslimit.smt_var(ssa)
 }
-pub fn chainid(ssa: &mut SSATracker) -> String {
+pub fn chainid(ssa: &mut SSATracker) -> SMTVariable {
     context().chainid.smt_var(ssa)
 }
-pub fn basefee(ssa: &mut SSATracker) -> String {
+pub fn basefee(ssa: &mut SSATracker) -> SMTVariable {
     context().basefee.smt_var(ssa)
 }
 
-pub fn calldataload(index: &String, ssa: &mut SSATracker) -> String {
+pub fn calldataload(index: SMTExpr, ssa: &mut SSATracker) -> SMTExpr {
     let calldata = context().calldata.smt_var(ssa);
     let arguments = (0..32)
```

Comment:
This could be
```
smt::concat((0..32).map((|i| smt::uf(calldata.clone(), [smt::bvadd(index.clone(), i)].iter()))
```

Reply:
if smt::concat would take an iterator instead of a slice

Reply:
hmm I can try to do that